autoload/airline/extensions/ale.vim

```diff
@@ -8,6 +8,25 @@ function! s:airline_ale_count(cnt, symbol)
   return a:cnt ? a:symbol. a:cnt : ''
 endfunction
 
+function! s:airline_ale_get_line_number(cnt, type) abort
+  if a:cnt == 0
+    return ''
+  endif
+
+  let l:open_lnum_symbol = get(g:, 'airline#extensions#ale#open_lnum_symbol', '(L')
+  let l:close_lnum_symbol = get(g:, 'airline#extensions#ale#close_lnum_symbol', ')')
+  let l:problem_type = (a:type ==# 'error') ? 'E' : 'W'
+  let l:problems = ale#engine#GetLoclist( bufnr('') )
+
+  for l:issues in l:problems
+    if (l:issues.type ==# l:problem_type)
```

Comment:
The parentheses are redundant here. You should write `let l:buffer = bufnr('')` above, and use `if l:issues.bufnr is l:buffer && l:issuers.type is# l:problem_type`. The loclist from ALE can contain problems for other files, so you can jump to them from the loclist.

The type may also be `'I'`, so you should consider handling that case.

Reply:
consider filtering the output of the `ale#engine#GetLoclist` directly, so that no loop is required to access the information. Also you can get rid of the l:problems variable.

Reply:
If you use `filter()`, make sure to use `copy()` first. The `List` returned is a reference.